packages/devtools_app/lib/src/framework/observer/_memory_web.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2025 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import 'package:web/web.dart';
+
+/// Estimates the memory usage of the DevTools web aplication, including all
+/// iFrames and workers.
+/// 
+/// See https://developer.mozilla.org/en-US/docs/Web/API/Performance/measureUserAgentSpecificMemory.
+Future<int?> measureMemoryUsageInBytes() async {
+  // Use of this API requires a secure context and cross origin isolation.
+  if (window.isSecureContext && window.crossOriginIsolated) {
+    final memory = await window.performance.measureUserAgentSpecificMemory();
+    return memory?.bytes;
+  }
+  return null;
+}
+
+extension PerformanceMemory on Performance {
```

Comment:
All the public things here (this extension, and the extension types), should probably be private.

Technically, by making them public we are actually exposing these as public API which means the API is different on web and other platforms. We don't want any user of this library to accidentally use these APIs.